### PR TITLE
Feature: Extra utility methods

### PR DESCRIPTION
**Pull request by maxelliott-causalens**
## Description and Motivation

- Added the `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph.get_nodes_for_variable_name` method to the  `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph` class.
- Added the `cai_causal_graph.graph_components.Node.is_source_node` and `cai_causal_graph.graph_components.Node.is_sink_node`
  methods to the `cai_causal_graph.graph_components.Node`, indicating whether the node is a source node or sink node
  respectively.

## How Has This Been Tested?

Added unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor (moving code around, general refactoring improvements)
- [ ] Documentation (documentation)

## PR Checklist:

- [X] I have reviewed my own PR? (review the PR as you are reviewing someone else's PR)
- [X] I have implemented all requirements? (see JIRA, project documentation)
- [ ] I am affecting someone else's work? If so: I have added those people as reviewers?
- [X] I have added comments to all the bits that are hard to follow
- [X] At a bare minimum, I tested this manually
- [X] I have added unit test(s)
- [ ] Is this a bugfix? If so have I added a regression test?
- [X] Does this PR satisfy the [one PR, one concern](https://medium.com/@fagnerbrack/one-pull-request-one-concern-e84a27dfe9f1) rule?
- [ ] If needed, documentation has been added/updated?
- [X] I have updated the appropriate changelog with a line for my changes
- [ ] If this PR breaks reproducibility, have I added a note in the changelog explaining the break in reproducibility
      and its extent?

### CODE REVIEW

File: cai_causal_graph/time_series_causal_graph.py [hunk 1]
```diff
@@ -1277,9 +1278,23 @@ def to_numpy_by_lag(self) -> Tuple[Dict[int, numpy.ndarray], List[str]]:
         return adjacency_matrices, self.variables
 
     def get_nodes_at_lag(self, time_lag: int = 0) -> List[TimeSeriesNode]:
-        """Return all nodes at time delta `time_lag`."""
+        """
+        Return all nodes at time delta `time_lag`.
+
+        :param time_lag: Tie lag to return nodes for. Default is `0`.
```
Comment (andrew-causalens) [inline cai_causal_graph/time_series_causal_graph.py:1284]: Tie -> Time

File: cai_causal_graph/time_series_causal_graph.py [hunk 2]
```diff
@@ -1277,9 +1278,23 @@ def to_numpy_by_lag(self) -> Tuple[Dict[int, numpy.ndarray], List[str]]:
...
+        """
+        # TODO Efficiency: don't loop through all nodes, by caching a lag -> node mapping: CAUSALAI-4384
         return [node for node in self.get_nodes() if node.time_lag == time_lag]
 
+    def get_nodes_for_variable(self, variable_name: str) -> List[TimeSeriesNode]:
```
Comment (andrew-causalens) [inline cai_causal_graph/time_series_causal_graph.py:1289]: Suggest `get_nodes_for_variable_name`

File: tests/test_time_series_graph.py
Comment (andrew-causalens) [inline tests/test_time_series_graph.py:1]: No unit test for the new `get_nodes_at_lag` method.

File: changelog.md [hunk 3]
```diff
@@ -4,6 +4,11 @@
 
 - Added the `cai_causal_graph.causal_graph.CausalGraph.replace_edge` method to the
   `cai_causal_graph.causal_graph.CausalGraph` class, which allows you to replace an existing edge with a new one.
+- Added the `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph.get_nodes_for_variable` method to the
```
Comment (andrew-causalens) [inline changelog.md:7]: Missing new `get_nodes_at_lag` method in changelog.

**Review by andrew-causalens — COMMENTED** [review]
Nice stuff @maxelliott-causalens! A few minor comments.

File: changelog.md (hunk 3 above)
Comment (maxelliott-causalens) [reply changelog.md:7]: This isn't a new method, I just updated the docsting to be consistent with my method

File: tests/test_time_series_graph.py
Comment (maxelliott-causalens) [reply tests/test_time_series_graph.py:1]: This isn't a new method, I just updated the docsting to be consistent with my method

File: changelog.md (hunk 3 above)
Comment (andrew-causalens) [reply changelog.md:7]: My bad. Completely missed that the `def` line wasn't new. Resolving this comment.

Comment (andrew-causalens) [inline changelog.md:7]: Need to update method name to `get_nodes_for_variable_name`

**Review by andrew-causalens — APPROVED** [review]
LGTM. Just small correction in changelog.

**Pull request merged**